server/webhook.go

```diff
@@ -199,3 +202,79 @@ func (p *Plugin) GetWebhookURL(jiraURL string, teamID, channelID string) (subURL
 
 	return subURL, legacyURL, nil
 }
+
+func (wh *webhook) getConnection(p *Plugin, instance Instance, notification webhookUserNotification) (con *Connection, err error) {
+	var mattermostUserID types.ID
+
+	// prefer accountId to username when looking up UserIds
+	if notification.jiraAccountID != "" {
+		mattermostUserID, err = p.userStore.LoadMattermostUserID(instance.GetID(), notification.jiraAccountID)
+	} else {
+		mattermostUserID, err = p.userStore.LoadMattermostUserID(instance.GetID(), notification.jiraUsername)
+	}
+	if err != nil {
+		return
+	}
+
+	// Check if the user has permissions.
+	con, err = p.userStore.LoadConnection(instance.GetID(), mattermostUserID)
+	if err != nil {
+		// Not connected to Jira, so can't check permissions
+		return
+	}
+
+	return
+}
+
+func (wh *webhook) checkWatcherUser(p *Plugin, instance Instance) {
+	watcherUsers := &[]jira.User{}
+	for _, notification := range wh.notifications {
+		c, err := wh.getConnection(p, instance, notification)
+		if err != nil {
+			continue
+		}
+		client, err2 := instance.GetClient(c)
+		if err2 != nil {
+			continue
+		}
+		watcherUsers, err = client.GetWatchers(wh.Issue.ID)
+		if err != nil {
+			continue
+		}
+		break
+	}
+
+	for _, watcherUser := range *watcherUsers {
+		whUserNotification := &webhookUserNotification{}
```

Comment:
We end up dereferencing later so we can just not use pointers in the first place

```suggestion
		whUserNotification := webhookUserNotification{}
```

Reply:
@mickmister I use a pointer to define `whUserNotification` because it is used as validation if the user is already in `wh.notifications` or not